components/eamxx/src/physics/rrtmgp/scream_rrtmgp_interface.cpp

```diff
@@ -878,5 +879,79 @@ namespace scream {
             });
         }
 
+        void compute_aerocom_cloudtop(
+            int ncol, int nlay, const real2d &tmid, const real2d &pmid,
+            const real2d &p_del, const real2d &z_del, const real2d &qc,
+            const real2d &qi, const real2d &cldfrac_tot, const real2d &nc,
+            real1d &T_mid_at_cldtop, real1d &p_mid_at_cldtop,
+            real1d &cldfrac_ice_at_cldtop, real1d &cldfrac_liq_at_cldtop,
+            real1d &cldfrac_tot_at_cldtop, real1d &nc_at_cldtop) {
+            // initialize the 1D "clear fraction" as 1 (totally clear) + and
+            // helper variable as 0
+            auto aerocom_clr = real1d("aerocom_clr", ncol);
+            auto aerocom_tmp = real1d("aerocom_tmp", ncol);
+            memset(aerocom_clr, 1);
+            memset(aerocom_tmp, 0);
+            // initialize the 2D "cloudy fraction" as 0 (totally clear)
+            auto aerocom_cld = real2d("aerocom_cld", ncol, nlay);
+            memset(aerocom_cld, 0);
+            // get gravity constant from constants
+            using physconst = scream::physics::Constants<Real>;
+            // Loop over all columns in parallel
+            yakl::fortran::parallel_for(
+                SimpleBounds<1>(ncol), YAKL_LAMBDA(int icol) {
+                  // Loops over all layers in serial (due to accumulative
+                  // product) Start at 2 (second highest) layer because the
+                  // highest one has no clouds
+                  for(int ilay = 2; ilay <= nlay; ++ilay) {
+                    // We only do the calculation if certain conditions are met
+                    if(qc(icol, ilay) + qi(icol, ilay) > 0 &&  // BAD_CONSTANT!
+                       cldfrac_tot(icol, ilay) > 0.001) {      // BAD_CONSTANT!
+                      aerocom_cld(icol, ilay) = cldfrac_tot(icol, ilay);
+                      aerocom_tmp(icol) =
+                          aerocom_clr(icol) *
+                          (1 - std::max(aerocom_cld(icol, ilay - 1),
```

Comment:
I am not sure this is standard compliant code, since std functions are in general not device friendly. I've seen some std::blah stuff compile and run, but I think it may be preferable to avoid them. An alternative option could be `ekat::impl::max(a,b)`.

Reply:
In this situations, if the code doesn't compile, it's usually in a GPU configuration. If you have access to a GPU-enabled system, you can build there to see whether things of this sort would cause problems. It is inconvenient, though, that the CPU and GPU variants of this dialect of "C++" require this sort of scrutiny.